src/rfbzero/experiment.py

```diff
@@ -426,8 +530,9 @@ def _validate_protocol(
         if cell_model.negative_concentrations():
             raise ValueError('Negative concentration detected')
 
-        def update_concentrations(i: float):
-            return cell_model.coulomb_counter(i, cls_degradation, ncls_degradation, crossover)
+        def update_concentrations(i: float) -> None:
+            """Performs coulomb counting, concentration updates via (optional) degradation and crossover mechanisms"""
```

Comment:
Seems odd to have a docstring for a local function. Maybe just have a normal comment?

Reply:
done